### ISSUE TITLE
Sprint - Aug 5 to Aug 16

### ISSUE THREAD

**Issue by daibhin** [PostHog/posthog#24105]
# Global Sprint Planning

## 3 things that might take us down

1. nothing

## Urgent incident follow-ups

https://github.com/orgs/PostHog/projects/103/views/2

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by daibhin** [comment]
# Team ~Replay~ David

**Support hero:** @daibhin

- @pauldambra - out 1st week
- @neilkakkar - out 1st week
- @daibhin - out the first two days
- @marandaneto - out 3 days

## Retro

Note: missed a lot of goals due to unexpected time off in the team

- @annikaschmid pricing changes were unexpectedly complicated. Almost over the line
  - retro scheduled
- @daibhin support load is ticking up again
  - look into support engineers handling proxy issues
  - update feature ownership for proxy related issues

### High priority

✅ done
🟢 in progress, on track
🟡 in progress, off track
🔴 missed

- 🟢 make sure pricing changes have landed @pauldambra
  - will be done by the end of the week, Raquel running last 3k customers today
- 🟡 error tracking
	- 🔴 if/when new ingestion pipeline
	  - we do have a new ingestion pipeline but no work done here
	  - paused until Paul is back and can catch up with Ben
	- 🟡 alerting/notifications? @daibhin 
	  - good chat with Ben and have a [clear path forward](https://posthog.slack.com/archives/C07AA937K9A/p1722419702909659)
	  - not implemented yet
- 🔴 heatmap ingestion separation? @pauldambra 

### Low priority / side quests

- 🔴 replay capture "large messages" @pauldambra 
- 🔴 feels like there is a person filtering bug where we don't get consistent results in replay @daibhin / @pauldambra
  - doesn't seem widespread, no other reports
- 🔴 replay bug investigation tooling @pauldambra / @daibhin 
	- playback snapshot by snapshot
	- chrome extension avoidance
- 🟢 session replays & survey following up on the new filtering experience @daibhin
  - replays looked positive, people filtered well
  - survey running, waiting on replies

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red

- 🟡  📱Goal 1: People think of PostHog as a mobile solution
- 🟢 🪲 Goal 2: Error tracking in people's hands
- 🟡 ⁉️ Goal 3: Hiring
  - Neil's joining the team!

### High priority

- Onboard @neilkakkar into the team
- Reduce support backlog (<=10 tickets, currently 25) @daibhin
- Error tracking alpha @daibhin 
  - no stack traces or alerting
  - JS SDK support only
  - convert fingerprints to arrays (currently strings)
  - Add alpha tags in UI and feedback banner
  - share with customers previously interviewed (no feature preview)
- Build out SDK support
  - Manual capture methods for JS and Python SDKs
- TBD: React Native session recording @marandaneto 

### Low priority / side quests

- replay bug investigation tooling @pauldambra 
	- playback snapshot by snapshot

**Comment by benjackwhite** [comment]
# Team Infra

## Retro / hangover

- 🟢  Get PoC of Warpstream for both Replay and events in place @frankh 
  - It's fully in production for all Replay things!
- 🔴  Actually implement reverse proxy sharding idea @frankh 
  - Warpstream took prio here (keeping things focused)
- 🟡 Helmfile is starting to hit its scaling limits and has random failures @frankh 
  - Made a change to improve some failures but its still not great
  - Also removed some other dependencies that were causing issues
- 🟢 Postgres EU schema maintenance @danielxnj 
  - Couple of bumps but we got there
- 🟢  K8s upgrades maintenance @danielxnj 
  - Karpenter upgrade still outstanding
- 🟢 RDS maintenance  @danielxnj 
- 🟢  Security group roll outs for locking down services @danielthedifficult 
  - Dev rollout being tested - looking good. Will carefully roll out to prod envs
- 🟢 Roll out celery keda to EU @benjackwhite 
- 🟡  Mr blobby autoscaling based on batch size utilzation @benjackwhite 
- 🟡  Infisical decision and plan @benjackwhite @danielxnj 
  - Cloud test worked fine
  - Deprioritized

## OKR

1.  🦹 Zero-trust security 🟢 
2. 🤓 10x Developer Experience 🟡 
3. 💪 Every service lives and dies alone
4. 💰 Save big on cost 🟡 


### High priority

- Warpstream analytics PoC 
  - 1. Get CDP using it @benjackwhite 
  - 2. Mirror capture traffic to it @danielxnj 
  - 3. Connect Clickhouse to it via entire Dev pipeline swapover @Daesgar @frankh
- Infisical decision @benjackwhite @danielxnj 
- Reverse proxy sharding implementation @frankh

**Comment by Phanatic** [comment]
# Team Feature Success

**Support hero:** @Phanatic 
Days off:
Juraj:  0 days
Phani: 0 days
Dylan: 0 days

## Retro

- event & action based triggers for surveys - actions - @Phanatic ✅ 
- no-code experiments RFC - @Phanatic ⌛ 
- Some experiments UX fixes (banners, display MDE next to the progress bar + explain), docs refresh - @jurajmajerik ✅ 
- Setup instrumentation for flip-flopping problem of experiment significance - @jurajmajerik 
- https://github.com/PostHog/posthog/issues/22131 continue on rewrite - @dmarticus ⌛ 

## Hang over items from previous sprint
- no-code experiments RFC - @Phanatic

---

## OKRs

1. Make sure feature flags can handle 10x current scale
2. No-code experiments
3. Split out experiments into its own product

### High priority

- Experiment confidence intervals @jurajmajerik
- verification framework/service to ensure that rust flags service works the same as decide. @dmarticus 
- no-code experiments RFC - @Phanatic 

### Low priority / side quests / maybe Neil will get to this next year

- Temporal queues for feature success - @Phanatic

**Comment by EDsCODE** [comment]
# Team Data <->
### OKR Q2 2024

### Objective

**Query 3000**

* **Key Results:**
    * Autocomplete
	* Increase general BI experience/product https://github.com/PostHog/meta/issues/157
	* Declutter the data warehouse UI and make the features intuitive to find

**Data Modeling MVP**

* **Key Results:**
	* Infrastructure decided and implemented
	* Integrating external data with feature flags
	* External data everywhere in insights/persons/cohorts
	* Get billing team to use modeling in posthog for their invoices_with_annual table
 
### Retro

- [ ] Data Modeling MVP/experimentation. We have some ideas that need to be tested @tomasfarias 
- [x] Data Warehouse Scene UI refactor/improvements @EDsCODE @Gilbert09 
	- [ ] https://docs.google.com/document/d/1Qy9e8BLMJKY7mmcrr1tv54mNuDc4cgVctwj_8kAhyec/edit 
- [ ] Get a better view of adoption both absolute and relative to broad posthog sign ups @EDsCODE 
- [x] BI improvements @Gilbert09 

### High Priorirty
- [ ] Data Modeling MVP/experimentation. We have some ideas that need to be tested @tomasfarias 
- [ ] Data warehouse design refresh @EDsCODE 
- [ ] BI continued https://github.com/PostHog/meta/issues/157 @Gilbert09

**Comment by raquelmsmith** [comment]
# Team Growth

## Retro
<details>  
<summary>Retro items</summary>

- @raquelmsmith (was out most of this week, didn't )
	- [ ] Launch / announce personless
	- [x] Help w/ session replay pricing update
    - [ ] Fix error that happens after creation, merge https://github.com/PostHog/posthog/pull/23069
    - [x] Finish teams-on-invites API changes
- @zlwaterfield (on call first first week - support second week)
	- [x]  Complete the subscribe to all products backfill (left over - ran into data issues that have been resolved)
	- [x]  Complete the startup plan metadata clean and dashboard (left over)
	- [ ]  Add at least one E2E SAML test (left over)
	- [x]  Improve activation error redirects in billing
	- [x]  Block customers from resubscribing if they've previously had their sub canceled from failed payments
	- [x]  Update support response time copy to use "target response time"
	- [ ]  Improve billing limits - store as number, improve validation / error handling in client and server
	- [x]  Startup customer events for customer.io emails when rolling off
	- [x] Misc plan issues - users on enterprise when they shouldn't be, mis matched tiers, free session replay plans, etc.
</details>  

## Q3 Goals
✅=finished 🟡=in progress 🔴=won't finish ⚪=not started
1. 🟡 Make onboarding awesome for Product analytics and Data warehouse (Raquel)
2. ⚪ Support self-serve annual commitments (Zach)
3. 🟡 Dive into the data to understand our billing metrics and customers better (Zach)
4. ✅ Launch pricing for data warehouse (Raquel)
5. 🟡 Hire 2 people (one for billing, one for auth/permissions focus)

## This sprint

Time off: @raquelmsmith (August 5-12)

- @zlwaterfield
	- [ ] teams annual plans
	- [ ] sentry profile / perf clean up
	- [ ] subscription cancellation reasoning dropdown
	- [ ] ICP score research - how is it calculated, where is it stored, when is it accessible, does it align with current views of ICP?
	- [ ] startup plan credits linked to infra cost
	- [ ] billing limit to numbers
	- [ ] SAML tests
- @raquelmsmith 
	- [ ] Support this week, Out next week 
	- [ ] Launch / announce personless
	- [ ] Fix shenanigans with session replay pricing update
	- [ ] Fix monthly enterprise customers revenue issue - https://posthog.slack.com/archives/C043VJ93L3B/p1721940308706669 
	- [ ] Set up billing alerts

**Comment by Twixes** [comment]
# Team Product Analytics

**Support hero:** @webjunkie (no secondary support)

## Retro

- 🍋‍🟩 @thmsobrmlr Remove `filters` from the frontend, allowing us to save new insights with `query` only.
- 🍋‍🟩 @skoob13 "First time this event was done by user" filer in Trends. Then, LLM-based insight generation MVP.
- 🍋‍🟩 @Twixes 1 week of support. Project environments rolled out internally. Onboarding @annaszell.
- 🟢 @aspicer Just 1 week of support.

## Goals

1. Rock-solid analytics (@thmsobrmlr  + @webjunkie  + @aspicer  + @anirudhpillai)
    1. 🟢 Legacy Minus – removing legacy insights code so that we can move fast
    1. 🔴 Tests Plus – shipping fewer bugs in the first place.
    1. 🔴 Metrics Plus – catching issues before before users report
    1. 🟡 Performance Plus - eliminating UX pain via maximum query performance/reliability, based on Metrics Plus data
    1. 🟢 Support Plus – sparking joy for users when they’re led to report a bug
1. Answering more product questions, deeper (@thmsobrmlr + @webjunkie + @aspicer + @anirudhpillai)
    1. 🔴 Growth Plus - increasing ease of onboarding, and subsequent retention
    1. 🟡 Analysis Plus - answering more product questions, more deeply
1. 🟡 ArtificialHog (@Twixes + @skoob13) – an LLM-based chat-like interface for answering product questions.

### High priority

Offsite week, including hackathon + hackathon followup!

- @thmsobrmlr Refactoring the backend from `Insight.filters` to `Insight.query` (dry run of insights table migration)

### Low priority / side quests

- @skoob13 End-to-ends for breakdowns

**Comment by fuziontech** [comment]
# Team Click Haus, Haus of the Hogs

## OKR Q2 2024

### Objective

**James as a Service -> Clickhouse as a Service**

- P0 tasks such as 
	- 🟡 Deletes
	- 🟢 Keeping clusters happy
	- 🟢 Provisioning more disks
	- 🟢 Schema Reviews
	- 🟢 Debugging
	- 🟡 Performance
	- 🟢 Backups/Restores
- Decide whether ByConity is the way forward
	- 🟢 Load it with data, set up 
	- 🟢 Test performance, test the functionality/compatibility gaps
- IF ByConity works, migrate over to it
	- 🟢  Enumerate all functionality that doesn’t work and update the functions/contribute to ByConity
	- 🟢 Syntax
	- 🟡 If it works on metal, put it in k8s with Karpenter
	- 🟡 Evaluate which nodes we should use
- IF ByConity doesn’t work, reshard US to look like EU cluster
	- 🟡 All clusters (Dev, US, EU) should be consistent in shape and topology. This will make it easier to manage and maintain the clusters and apply learnings from one cluster to another.
	- 🟢 We want all cluster operations to be automated and managed through some form of infra as code that is available in source control.
	- 🟡 Schema management on ClickHouse should be entirely automated and managed through source control with no exceptions. This includes Coordinator schemas.
	- 🟢 We should be able to spin up and down replicas of any cluster with no manual intervention.
	- 🟢 We should be able to upgrade ClickHouse versions with no manual intervention.
	- 🟡 We should have tooling / runbooks for resharding (if we continue down the current coordinator path)

## Board
https://github.com/orgs/PostHog/projects/85/views/2

## Retro

@Daesgar - Somewhat hard emotionally for me because of everything going on (with being only person in EU for CH). Incident we had was pretty awesome facing this kind of thing and dealing with it/recovering from it. Not best to have this kind of incident, but you learn a lot from them. Validated connection at the pueblo ✅ 

@fuziontech - It was good to have an incident like last week so that we know that were never losing data. We understand the problem better and how to address it in the future. Basically the system is Anti-Fragile. Slowly our systems will become more robust. Overall good week. I wish we still had more time because distractions like this do slow down progress against the planned work. Sudden reprioritization of work.

## Board Snapshot 
<img width="1455" alt="image" src="https://github.com/user-attachments/assets/81f66d00-2d74-4ea2-ac03-f6403abb5bf8">

**Comment by robbie-c** [comment]
# Team web analytics

**Support hero:** @robbie-c 

## Retro

This sprint ended up being quite different from what was planned, and I spent a lot of it in zendesk.

The session table v2 roll out had to be reverted due to a problem with the backfill, but this is fixed and rolled out again and looks good.

#### Planned Tasks
🟡 Figure out and fix [#23690 Person profiles not being merged, using recent versions of the JS sdk](https://github.com/PostHog/posthog/issues/23690) (maybe 2 different issues, customer facing one is more urgent than posthog facing one)
🟢 Get everyone in EU onto sessions v2, iron out any problems
🔴 Come up with a solution for a few linked issues: people want a default set of filters, and people want it to be easier to separate their sites. There's been a lot of dicussion on this in the past, see https://github.com/PostHog/posthog/issues/18863 https://github.com/PostHog/posthog/issues/12181 https://github.com/PostHog/posthog/issues/20314 <-- I'm leaning towards the first link here but figuring this out is part of the work

## OKR

1. Make querying fast enough for large customers
2. Heavily requested features
3. Work better with other products
4. Product and growth

### High priority
- Figure out what's going on with cross domain tracking problems (e.g. https://github.com/PostHog/posthog/issues/23690 https://posthog.slack.com/archives/C05LJK1N3CP/p1720719062387739 )
- (from last sprint) Come up with a solution for a few linked issues: people want a default set of filters, and people want it to be easier to separate their sites. There's been a lot of dicussion on this in the past, see https://github.com/PostHog/posthog/issues/18863 https://github.com/PostHog/posthog/issues/12181 https://github.com/PostHog/posthog/issues/20314 <-- I'm leaning towards the first link here but figuring this out is part of the work

### Stretch goals
- Add some visual regression tests for web analytics
- Docs changes for the improvements to attribution

### Ongoing
- In the background, continue to backfill the sessions table

**Comment by benjackwhite** [comment]
# Team CDP 

Off: Brett 1 week
Support: Oliver

## Retro

### High priority

- 🔴 Phase out final uses for jobs and scheduler deployments
- 🟢  Migrate all semver flattener users to use the inlined plugin and inline some more plugins (user agent done too)
- 🟢 Rusty hook to use app metrics v2 for debuggability (Brett)
- 🟢 New Pipelines UI is fully out
- 🟢 Support for AWS kinesis destination (Hog STL)
- 🟢 Throughput estimation graph
- 🟢 App metrics for hog functions
- 🟢 Native oauth flows for 
- 🟢  Hook up to an actual webhook service
- 🟡  mvp of event and property definitions as a separate service (Oliver)

## OKR
(to be refactored)

### High priority

- Property/Event definitions off the critical path @oliverb123 
- Exception ingestion @bretthoerner @benjackwhite 
- Delivery service V0.2
  - Stress test the existing rusty hook system and any improvements for a V0.1
  - Architect V0.2 ("Codename Cyclotron") 
- Final UX stuff to enable releasing this in some form @benjackwhite 

### Side quests
- Warpstream for events testing (Mostly done by infra)

**Comment by kitsirota** [comment]
@Twixes, @daibhin, @benjackwhite, @Phanatic

We are disappointed to see that the following request has been marked as "Low priority / side quests":
-@skoob13: End-to-ends for breakdowns

This issue has been ongoing since 2020 and is critical for enabling marketing teams to transition to PostHog. For reference, please see the original issue: [PostHog Issue #938](https://github.com/PostHog/posthog/issues/938).

Can we escalate this work?

Multiple breakdowns are the most important feature preventing marketing teams from moving to PostHog from Google Analytics (GA). Resolving this issue would be the single biggest improvement to the platform, enabling the porting of reports from GA to PostHog.

**Comment by kitsirota** [comment]
@Twixes, @daibhin, @benjackwhite, @Phanatic

We are disappointed to see that the following request has been marked as "Low priority / side quests":
-@skoob13: End-to-ends for breakdowns

This issue has been ongoing since 2020 and is critical for enabling marketing teams to transition to PostHog. For reference, please see the original issue: [PostHog Issue #938](https://github.com/PostHog/posthog/issues/938).

Can we escalate this work?

Multiple breakdowns are the most important feature preventing marketing teams from moving to PostHog from Google Analytics (GA). Resolving this issue would be the single biggest improvement to the platform, enabling the porting of reports from GA to PostHog. I can't think of any reason to prevent marketing teams from making this transition.

**Comment by kitsirota** [comment]
@Twixes, @daibhin, @benjackwhite, @Phanatic

It is very disappointing to see that the following request has been marked as "Low priority / side quests":
@skoob13: End-to-ends for breakdowns

This feature gap has been ongoing since 2020 and is critical for enabling marketing teams to transition to PostHog. For reference, please see the original issue: [PostHog Issue #938](https://github.com/PostHog/posthog/issues/938).

Can we escalate this work?

Multiple breakdowns are the most important feature preventing marketing teams from moving to PostHog from Google Analytics (GA). Resolving this issue would be the single biggest improvement to the platform, enabling the porting of reports from GA to PostHog. I can't think of any reason to prevent marketing teams from making this transition.
